packages/exchange/src/pods/demand/demand-time-period.service.ts

```diff
@@ -18,9 +18,9 @@ export class DemandTimePeriodService {
         const range = moment.range(start, end);
 
         const { diff, step } = this.timeFrameToTimeDiff(createDemand.periodTimeFrame);
-        const validFrom = Array.from(range.by(diff, { step, excludeEnd: true })).map((v) =>
-            v.toDate()
-        );
+        const validFrom = Array.from(
+            range.by(diff, { step, excludeEnd: createDemand.excludeEnd })
```

Comment:
Since createDemand.excludeEnd would be marked as optional, please change:

`excludeEnd: createDemand.excludeEnd`

to

`excludeEnd: createDemand.excludeEnd ?? true`